### ISSUE TITLE
Add "Parent" level to Projects

### ISSUE THREAD

**Issue by pzljanb** [Puzzlepart/did#1245]
- [x] I have searched for existing issues to avoid creating duplicates

### TLDR
Parent Node to be implemented to give grouping of Projects related to the same delivery/ contract

### Description

* What is the feature?
A new level in the Customer / Project hierarchy,

* Why should we add this feature?
To give grouping of a delivery/ contracts total timeconsumption separated within a Customers totals.

* How does this improve the product?
Gives Project Managers (and others) control of the total within a delivery/contract

Describe the functionality in as much detail as you can, feel free 
to add sketches and screenshots.

New entitiy "Contract"
Contract is to be owned by Customer object, and created on Customer level.
Create relation (1:n) between Contract and Project
From Project entity contract could be selected among the ones owned by the same customer.
(Select project from Contract view?)

Properties on Contract entity:
- Name (Exists in CRM)
- Description (Exists in CRM)
- Contract ID (Exists in CRM)
- Contract owner (Exists in CRM)

Enhancement to Project entity:
Rename to "Activity"
Add property: Hour rate
Add property: Fixed price
Add property: Invoice type ; values: [by hours], [Fixed Price]

**Comment by olemp** [comment]
@okms @pzljanb Might need to redo the description for this one?

First _Customer / Project hierarchy_ is mentioned: 

![image](https://github.com/user-attachments/assets/fa56d744-3347-4952-a402-0c28ac4bf900)

Then renaming of Project to Activity:

![image](https://github.com/user-attachments/assets/7b7ae96f-480b-4520-a1e7-656099b52bab)

Then these coments from @okms:

![image](https://github.com/user-attachments/assets/91fa44bf-bbad-4778-abc7-4d26b673b4ec)


What exactly do we want to do? Make adjustments to the roles implementation in https://didapp-dev.azurewebsites.net/? Or add contract entity, and rename project to activity etc?

**Comment by pzljanb** [comment]
Add parameter "Parent Project" to Projects.
Should be selectable from Projectproperties.

Include information in Excelreports and on Projectproperties in did.
did project listing must include "Parent project" column

**Comment by olemp** [comment]
@pzljanb Dette kan du teste nå.

Se også notater fra @damsleth sin testing i https://github.com/Puzzlepart/did/pull/1250
